design/ceph/submariner-globalnet.md

```diff
@@ -0,0 +1,96 @@
+# Submariner Globalnet support in Rook
+
+
+## Use case
+As a rook-ceph user, I should be able to setup mirroing across multiple clusters with overlapping networks in order to safegurd my data in case of any disaster.
+
+
+## Background
+[Submariner](https://submariner.io/getting-started/) provides cross-cluster connectivity between two kubernetes clusters when they have different CIDRs. Many a times customers don't have control over the network configuration of the clusters, resulting in clusters with overlapping CIDRs (ServiceCIDR and ClusterCIDR). For such scenarios, Submariner provides another tool called [Globalnet](https://submariner.io/getting-started/architecture/globalnet/). It provides cross-cluster connectivity from pods to remote services using their global IPs.
+
+In order to support kubernetes clusters connected using Submariner-Globalnet, Rook needs to use the global IPs provided by Globalnet instead of the service cluster IP.
+
```

Comment:
Add background here that addresses that we don't want to hard-code support for only Submariner. 

Submariner uses the upstream Multi-Cluster Services (MCS) API which is being defined by the K8s SIG “mcs-api”. This is also associated with the K8s SIG community “sig-multicluster”.
MCS API SIG GitHub - https://github.com/kubernetes-sigs/mcs-api

Reply:
While we don't want the implementation specific to Submariner, how about adding details about what we have observed about Submariner behavior that is important to consider for any implementation? For example:
- The traffic in-cluster to mons and OSDs is observed to be routed directly to the mons or OSDs. This is very important so traffic is not routed through a single bottleneck.
- The traffic across cluster is all routed through the submariner gateway. This known to be a bottleneck, but is necessary for the tunnel.